execution/commitment/commitmentdb/commitment_context.go

```diff
@@ -43,21 +43,136 @@ type sd interface {
 	CommitmentCapture() bool
 }
 
-type SharedDomainsCommitmentContext struct {
-	sharedDomains      sd
-	updates            *commitment.Updates
-	patriciaTrie       commitment.Trie
-	justRestored       atomic.Bool // set to true when commitment trie was just restored from snapshot
-	trace              bool
+type StateReader interface {
```

Comment:
My original idea was to reuse the `StateReader` interface that's most commonly used:

```Go
type StateReader interface {
	ReadAccountData(address common.Address) (*accounts.Account, error)
	ReadAccountDataForDebug(address common.Address) (*accounts.Account, error)
	ReadAccountStorage(address common.Address, key common.Hash) (uint256.Int, bool, error)
	HasStorage(address common.Address) (bool, error)
	ReadAccountCode(address common.Address) ([]byte, error)
	ReadAccountCodeSize(address common.Address) (int, error)
	ReadAccountIncarnation(address common.Address) (uint64, error)

	SetTrace(trace bool, tracePrefix string)
}
```

Now we are introducing yet another StateReader interface, and not reusing what already exists, which adds extra maintenance costs.

Reply:
Same as above: IMO this can be part of another PR, if agreed (I think it can be questionable if the same `StateReader` must be used everywhere or if `commitmentdb` deserves its own `StateReader` abstraction).
The refactoring proposed in this PR is quite straightforward and even naive, but helps to better reasoning about state access in commitments, so I would got for it as is and then iterate further.